shared_file_storage/main.tf

```diff
@@ -0,0 +1,14 @@
+resource "azurerm_storage_account" "storage" {
```

Comment:
I think to your point above that since different storage accounts could have different replication types and  it might make sense to associate the storage account with the file share.  It's likely based on what you plan to create the file share for, might help make the decision of what kind of storage account you want.  For that reason I think it makes more sense to associate it.  Let me know how you want to proceed.